ui/job-view/PushJobs.jsx

```diff
@@ -187,46 +192,33 @@ export default class PushJobs extends React.Component {
     findJobInstance(job.id, false).toggleRunnableSelected();
   }
 
-  filterPlatform(platform, selectedJobId) {
-    platform.visible = false;
-    platform.groups.forEach((group) => {
-      group.visible = false;
-      group.jobs.forEach((job) => {
-        job.visible = this.thJobFilters.showJob(job) || job.id === selectedJobId;
-        if (job.state === 'runnable') {
-          job.visible = job.visible && this.props.push.isRunnableVisible;
-        }
-        job.selected = this.$rootScope.selectedJob ? job.id === this.$rootScope.selectedJob.id : false;
-        if (job.visible) {
-          platform.visible = true;
-          group.visible = true;
-        }
-      });
-    });
-    return platform;
-  }
-
   filterPlatformCallback(platform, selectedJobId) {
-    this.filterPlatform(platform, selectedJobId);
-    if (Object.keys(this.state.platforms).length) {
-      this.setState({ platforms: { ...this.state.platforms } });
+    const { push, filterModel } = this.props;
+    const { platforms } = this.state;
+
+    // This actually filters the platform in-place.  So we just need to
+    // trigger a re-render by giving it a new ``platforms`` object instance.
+    PushJobs.filterPlatform(platform, selectedJobId, push, filterModel);
+    if (platforms.length) {
+      this.setState({ platforms: [...platforms] });
     }
   }
 
   render() {
-    const platforms = this.state.platforms || {};
-    const { $injector, repoName } = this.props;
+    const platforms = this.state.platforms || [];
+    const { $injector, repoName, filterModel } = this.props;
 
     return (
       <table id={this.aggregateId} className="table-hover" data-job-clear-on-click>
         <tbody onMouseDown={this.onMouseDown}>
-          {platforms ? Object.keys(platforms).map(id => (
-          platforms[id].visible &&
+          {platforms ? platforms.map(platform => (
+          platform.visible &&
           <Platform
-            platform={platforms[id]}
+            platform={platform}
             repoName={repoName}
             $injector={$injector}
-            key={id}
+            key={platform.id}
+            filterModel={filterModel}
             filterPlatformCb={this.filterPlatformCallback}
           />
         )) : <tr>
```

Comment:
nit: Do you need to have an empty `<tr>` or could you do `{platforms &&` rather than the ternary?